blog/03-04-2023-creating-a-twitter-like-decentralized-application-on-celo-using-solidity/creating-a-twitter-like-decentralized-application-on-celo-using-solidity.md

```diff
@@ -0,0 +1,365 @@
+
+At Last, we need to return `true` because we have successfully created a tweet in the `tweets` state array variable like database and made a Transaction Log on the Transaction block that a Tweet had been created so at the end we ```return true```. 
+
+## Getting the Tweet which is recorded on the Celo blockchain network 
+
+For reading the tweets from the Celo blockchain network. We are creating two function named ```getAllTweets``` and ```getTweetsByUser```. 
+
+```solidity
+
+ function getAllTweets() public view returns(Tweet [] memory){ 
+
+     return tweets; 
+
+ } 
+
+ ``` 
+
+We have created the `getAllTweets` function that takes no parameter and is a public function, which can be called with the contract itself and have a view modifier which means no state variable will be modified and can read the state variable and will return Tweet array variable which is created to store all the Tweet. At the end of `getAllTweets` function, the function returns the tweets state variable which acts as a storage for this contract. 
+
+```solidity
+
+    function getTweetByUser(address _sender) public view returns(Tweet [] memory){
+
+        uint tweetsLength = Tweets.length;
+
+        uint _userTweetCount = userTweetCount[_sender];
+
+        Tweet [] memory _tweet = new Tweet[](_userTweetCount);
+
+        uint j;
+
+        for(uint a = 0; a < tweetsLength; a++){
+
+            if(Tweets[a].tweeter == _sender){
+
+                _tweet[j] = Tweets[a];
+
+                j += 1;
+            }
+        }
+
+        return _tweet;
+    }
+
+ ``` 
+
+In the `getTweetByUser` function we take a parameter `_sender` as a tweeter address which will then return an array of tweets, the tweeter created. Firstly, we create two variables `tweetsLength` and `_userTweetCount` to respectively store the current total number of tweets in our smart contract and the total number of tweets made by the `_sender`. We then initialize the `_tweet` variable which we will use to temporarily store all the tweets made by the `_sender` address. Next, we loop through the `Tweets` array, and for each iteration, we check if the tweet is owned by the `_sender` and if it is, we store it in the `_tweet` array. Finally, we return the `_tweet` array.
+
+## Final Source Code 
+
+Here is your final Source code for this first series of Twitter Dapps. 
+
+```solidity
+
+//SPDX-License-Identifier: MIT
+pragma solidity = 0.8.9;
```

Comment:
It will throw error , so use -- pragma solidity ^0.8.9;

I am making this change for now, make sure this does not happen in future